Tests/Brokerages/OrderCrossingBrokerageTests.cs

```diff
@@ -223,13 +294,36 @@ private static PhonyBrokerage InitializeBrokerage(params (string ticker, decimal
             return brokerage;
         }
 
+        /// <summary>
+        /// Asserts that the brokerage account holdings are not empty and that the first holding has a positive quantity.
+        /// </summary>
+        /// <param name="brokerage">The brokerage instance to check the account holdings.</param>
+        /// <exception cref="AssertionException">
+        /// Thrown if the account holdings are empty or the first holding has a non-positive quantity.
+        /// </exception>
         private static void AssertThatHoldingIsNotEmpty(IBrokerage brokerage)
         {
             var holdings = brokerage.GetAccountHoldings();
             Assert.Greater(holdings.Count, 0);
             Assert.Greater(holdings[0].Quantity, 0);
         }
 
+        /// <summary>
+        /// Asserts that an order with the specified status has arrived by waiting for an event signal.
+        /// </summary>
+        /// <param name="resetEvent">The event to wait on, which signals that an order status update has occurred.</param>
+        /// <param name="brokerage">The phony brokerage instance used to check the order status.</param>
+        /// <param name="comingOrderStatus">The expected status of the coming order to assert.</param>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if there is not exactly one order with the specified status.
+        /// </exception>
+        private static void AssertComingOrderStatusByEvent(AutoResetEvent resetEvent, PhonyBrokerage brokerage, OrderStatus comingOrderStatus)
+        {
+            resetEvent.WaitOne();
```

Comment:
Add `Assert.False(resetEvent.WaitOne(TimeSpan.FromSeconds(5)))` so it failes on timeout, should not `Wait()` without timeout review other cases